packages/expo-file-system/ios/EXFileSystem/EXSessionTasks/EXSessionDownloadTaskDelegate.m

```diff
@@ -0,0 +1,60 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXFileSystem/EXSessionDownloadTaskDelegate.h>
+#import <EXFileSystem/NSData+EXFileSystem.h>
+
+@interface EXSessionDownloadTaskDelegate ()
+
+@property (strong, nonatomic) NSURL *localUrl;
+@property (nonatomic) BOOL shouldCalculateMd5;
+
+@end
+
+@implementation EXSessionDownloadTaskDelegate
+
+- (instancetype)initWithResolve:(UMPromiseResolveBlock)resolve
+                         reject:(UMPromiseRejectBlock)reject
+                       localUrl:(NSURL *)localUrl
+             shouldCalculateMd5:(BOOL)shouldCalculateMd5
+{
+  if (self = [super initWithResolve:resolve reject:reject])
+  {
+    _localUrl = localUrl;
+    _shouldCalculateMd5 = shouldCalculateMd5;
+  }
+  return self;
+}
+
+- (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTask *)downloadTask didFinishDownloadingToURL:(NSURL *)location
+{
+  NSError *error;
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+  if ([fileManager fileExistsAtPath:_localUrl.path]) {
+    [fileManager removeItemAtURL:_localUrl error:&error];
+    if (error) {
+      self.reject(@"ERR_FILE_SYSTEM_UNABLE_TO_REMOVE",
+                  [NSString stringWithFormat:@"Unable to remove file from local URI: '%@'", error.description],
+                  error);
+      return;
+    }
+  }
+
+  [fileManager moveItemAtURL:location toURL:_localUrl error:&error];
+  if (error) {
+    self.reject(@"ERR_FILE_SYSTEM_UNABLE_TO_SAVE",
+                [NSString stringWithFormat:@"Unable to save file to local URI: '%@'", error.description],
+                error);
+    return;
+  }
+
+  NSMutableDictionary *result = [[EXSessionTaskDelegate parseServerResponse:downloadTask.response] mutableCopy];
```

Comment:
An refactor opportunity would be to make the `parseServerResponse` an instance method and override it in this class. Basically what we're doing here is adding class-specific properties to a more generic object, right? What do you think? (There is a non-zero possibility I'm overrefactoring this 🤔 )